kartothek/core/dataset.py

```diff
@@ -311,6 +311,20 @@ def query(self, indices: List[IndexBase] = None, **kwargs) -> List[str]:
         for column, value in kwargs.items():
             if column in combined_indices:
                 candidate_set &= set(combined_indices[column].query(value))
+            else:
+                if (
+                    # FIXME: `self.tables[0]`. What if column exists in another table?
+                    not self.table_meta[self.tables[0]]
+                    .internal()
+                    .field_by_name(column)
+                ):
+                    raise ValueError(
+                        f"Column provided (`{column}`) does not appear in dataset"
+                    )
+                # Rather than returning all partitions, we abort the execution at this stage to be safe
+                raise ValueError(
```

Comment:
_Technically_ this is a breaking change in behaviour :/
I do agree that this is a sane behaviour, though.

Reply:
I would actually propose to add a _new_ function / functionality to query by predicates (essentially call dispatch_metapartitions but only return the partition names). This would amount to the same functionality and would allow to deal with this in a non-breaking way, especially since I would like to get rid of the query-by-dict syntax in general